Userland/Libraries/LibHID/ReportDescriptorParser.cpp

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2025, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringBuilder.h>
+#include <LibHID/ReportDescriptorParser.h>
+
+namespace HID {
+
+#ifndef KERNEL
+ErrorOr<void> dump_report_descriptor(ReadonlyBytes report_descriptor)
```

Comment:
(I think the more common approach is to dump the internal representation that your normal parsing routine produces, so that you don't have two parsing paths that can get out of sync)